moveit_setup_assistant/CMakeLists.txt

```diff
@@ -1,41 +1,33 @@
 cmake_minimum_required(VERSION 3.10.2)
 project(moveit_setup_assistant)
 
-if(NOT "${CMAKE_CXX_STANDARD}")
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 14)
 endif()
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
-if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-  # suppress warning in Ogre
-  add_compile_options(-Wno-deprecated-register)
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
-# definition needed for boost/math/constants/constants.hpp included by Ogre to compile
-add_definitions(-DBOOST_MATH_DISABLE_FLOAT128)
+# find dependencies
+find_package(ament_cmake REQUIRED)
+find_package(moveit_core REQUIRED)
+find_package(moveit_ros_planning REQUIRED)
+find_package(moveit_ros_visualization REQUIRED)
+# find_package(rosconsole REQUIRED)
+find_package(rviz2 REQUIRED)
+find_package(rviz_common REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(urdf REQUIRED)
+find_package(srdfdom REQUIRED)
 
-if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
-  set(CMAKE_BUILD_TYPE Release)
-endif()
-
-find_package(Boost REQUIRED COMPONENTS thread filesystem system program_options)
-find_package(catkin REQUIRED COMPONENTS
-  moveit_core
-  moveit_ros_planning
-  moveit_ros_visualization
-  rviz
-  rosconsole
-  roscpp
-  urdf
-  srdfdom
-)
 find_package(ompl REQUIRED)
-include_directories(SYSTEM ${catkin_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS} ${OMPL_INCLUDE_DIRS})
-include_directories(include)
 
 # Qt Stuff
-if(rviz_QT_VERSION VERSION_LESS "5")
+if(rviz_QT_VERSION AND rviz_QT_VERSION VERSION_LESS "5")
```

Comment:
As far as I know, we no longer depend on Qt4 so this should be removed

Reply:
QT5 was released on 2012 so I dropped support for older versions.